pkg/api/v1beta3/types.go

```diff
@@ -866,6 +867,8 @@ type PodSpec struct {
 	DNSPolicy DNSPolicy `json:"dnsPolicy,omitempty" description:"DNS policy for containers within the pod; one of 'ClusterFirst' or 'Default'"`
 	// NodeSelector is a selector which must be true for the pod to fit on a node
 	NodeSelector map[string]string `json:"nodeSelector,omitempty" description:"selector which must match a node's labels for the pod to be scheduled on that node"`
+	// Optional: Conflicts are used to identify pods that cannot be scheduled together.
+	Conflicts []labels.LabelSelector `json:"conflicts,omitempty"`
```

Comment:
In the API we need to use a type with proper json and description tags. This also needs a description tag.

Please see #341 and #7053 for more details.